os/include/tinyara/common_logs/common_logs.h

```diff
@@ -0,0 +1,82 @@
+typedef enum
+{
+	/* Memory Operations */
+	COMMON_LOG_ALLOC_FAIL,
+	COMMON_LOG_NULL_CHECK_FAIL,
+	COMMON_LOG_OUT_OF_MEMORY,
+	COMMON_LOG_MEM_LEAK,
+	COMMON_LOG_BUFFER_OVERFLOW,
+
+	/* File Operations */
+	COMMON_LOG_FILE_NFOUND,
+	COMMON_LOG_FILE_OPEN_ERROR,
+	COMMON_LOG_FILE_READ_ERROR,
+	COMMON_LOG_FILE_WRITE_ERROR,
+	COMMON_LOG_FILE_SEEK_ERROR,
+	COMMON_LOG_FILE_IOCTL_ERROR,
+
+	/* Network Operations */
+	COMMON_LOG_CON_TIMEOUT,
+	COMMON_LOG_CON_REFUSED,
+	COMMON_LOG_DATA_TRANS_ERROR,
+	COMMON_LOG_HANDSHAKE_FAILED,
+	COMMON_LOG_CERTIFICATE_INVALID,
+
+	/* Authentication and Authorization */
+	COMMON_LOG_INVALID_CREDENTIALS,
+
+	/* Commonly used logs for printing information */
+	COMMON_LOG_VALUE_OF,
+	COMMON_LOG_INVALID_VAL,
+	COMMON_LOG_FAILED_OP,
+	COMMON_LOG_PERFORMING_OP,
```

Comment:
can have a function to retrieve the sting which can also validate the array out of bound. Also can we shorten the enum names?

Reply:
I didnt understand the second part, what do you mean by validate out of bounds?

Reply:
i mean array out of bound. in the usage.

Reply:
can you give an example? what do you mean by validate array out of bound? is it printing the pointer of the array?

Reply:
I updated the names in enum.